platform/saivpp/vpplib/SwitchStateBaseFdb.cpp

```diff
@@ -1082,3 +1110,264 @@ sai_status_t SwitchStateBase::vpp_delete_bvi_interface(
+	return ~0;
+    }
+    return it->second;
+}
+
+int SwitchStateBase::remove_lag_to_bond_entry(const sai_object_id_t lag_oid)
+{
+   auto it = m_lag_bond_map.find(lag_oid);
+
+    if (it == m_lag_bond_map.end())
+    {
+        SWSS_LOG_ERROR("failed to find lag swif index for : %s", sai_serialize_object_id(lag_oid).c_str());
+        return ~0;
+    }
+
+    SWSS_LOG_NOTICE("Removing lag object swif index: %s", sai_serialize_object_id(lag_oid).c_str());
+    m_lag_bond_map.erase(it);
+    return 0;
+}
+
+sai_status_t SwitchStateBase:: createLag(
+        _In_ sai_object_id_t object_id,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list)
+{
+    SWSS_LOG_ENTER();
+
+    auto sid = sai_serialize_object_id(object_id);
+    CHECK_STATUS(create_internal(SAI_OBJECT_TYPE_LAG, sid, switch_id, attr_count, attr_list));
+    return vpp_create_lag(object_id, attr_count, attr_list);
+
+}
+
+sai_status_t SwitchStateBase::vpp_create_lag(
+	_In_ sai_object_id_t lag_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list)
+{
+    uint32_t bond_id, mode, lb;
+    uint32_t swif_idx = ~0;
+    const char *hw_ifname;
+    SWSS_LOG_ENTER();
+
+    //set mode and lb. ONiC config does not have provision to pass mode and load balancing algorithm
+    mode = VPP_BOND_API_MODE_ROUND_ROBIN;
+
+    //if LACP is to be enabled in VPP
+    //mode = VPP_BOND_API_MODE_LACP;
+    lb = VPP_BOND_API_LB_ALGO_L2;
+    bond_id = ~0;
+
+    create_bond_interface(bond_id, mode, lb, &swif_idx);
+    SWSS_LOG_NOTICE("vpp bond interfae created if index:%d\n", swif_idx);
+    //update the lag to bond map
+    m_lag_bond_map[lag_id] = swif_idx;
+    refresh_interfaces_list();
+
+    // Set the bond interface state up
+    hw_ifname = vpp_get_swif_name(swif_idx);
+    SWSS_LOG_NOTICE("Setting lag hw interface state to up :%s",hw_ifname);
+    interface_set_state(hw_ifname, true);
+    return SAI_STATUS_SUCCESS;
+}
+
+sai_status_t SwitchStateBase:: removeLag(
+        _In_ sai_object_id_t lag_oid)
+{
+    SWSS_LOG_ENTER();
+
+    vpp_remove_lag(lag_oid);
+    auto sid = sai_serialize_object_id(lag_oid);
+    CHECK_STATUS(remove_internal(SAI_OBJECT_TYPE_LAG, sid));
+    return SAI_STATUS_SUCCESS;
+}
+
+sai_status_t SwitchStateBase::vpp_remove_lag(
+        _In_ sai_object_id_t lag_oid)
+{
+    SWSS_LOG_ENTER();
+
+    uint32_t lag_swif_idx = lag_to_bond_if_idx(lag_oid);
+    SWSS_LOG_NOTICE("lag swif idx :%d",lag_swif_idx);
+    auto lag_ifname =  vpp_get_swif_name(lag_swif_idx);
+    SWSS_LOG_NOTICE("lag swif idx :%d swif_name:%s",lag_swif_idx, lag_ifname);
+    if (lag_ifname == NULL)
+    {
+        SWSS_LOG_NOTICE("LAG interface name is not found for LAG PORT :%s",sai_serialize_object_id(lag_oid).c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    //Delete the Bond interface
+    delete_bond_interface(lag_ifname);
+    remove_lag_to_bond_entry(lag_oid);
+
```

Comment:
I think you need to call refresh_interfaces_list() to build new interface id to name mapping once bond interface is deleted.

Reply:
Yes, done